predicators/structs.py

```diff
@@ -1409,13 +1409,39 @@ def ground(self, objects: Tuple[Object]) -> _GroundLDLRule:
 
     @cached_property
     def _str(self) -> str:
-        nsrt_param_str = ", ".join([str(v) for v in self.nsrt.parameters])
-        return f"""LDLRule-{self.name}:
-    Parameters: {self.parameters}
-    Pos State Pre: {sorted(self.pos_state_preconditions, key=str)}
-    Neg State Pre: {sorted(self.neg_state_preconditions, key=str)}
-    Goal Pre: {sorted(self.goal_preconditions, key=str)}
-    NSRT: {self.nsrt.name}({nsrt_param_str})"""
+        parameter_str = "(" + " ".join(
+            [f"{p.name} - {p.type.name}" for p in self.parameters]) + ")"
+
+        def _atom_to_str(atom: LiftedAtom) -> str:
+            args_str = " ".join([v.name for v in atom.variables])
+            return f"({atom.predicate.name} {args_str})"
+
+        inner_preconditions_strs = [
+            _atom_to_str(a) for a in sorted(self.pos_state_preconditions)
```

Comment:
I think leaving it sorted is the right way to go, but just wanted to make a note in case future work has some inconsistency with ordering (I bumped into this once)